src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -131,6 +132,8 @@ export function DiscoverMainRoute({ customizationCallbacks, isDev }: MainRoutePr
 
   const loadSavedSearch = useCallback(
     async (nextDataView?: DataView) => {
+      if (!isCustomizationServiceInitialized) return;
```

Comment:
I'm trying now a different approach to make more explicit the steps taken to initialize the customizations, I'll push the changes as soon as everything looks good.

Reply:
This approach seems reasonable to me -- Discover probably shouldn't start loading anything until customizations have finished initializing since they may want to modify initial state like you're doing. I'm also open to more explicit steps though, so I'll check back in on the new approach once pushed.